hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/StructuredTraceGraphBuilder.java

```diff
@@ -24,7 +24,7 @@ public static StructuredTraceGraph buildGraph(StructuredTrace trace) {
     if (GraphBuilderUtil.isDifferentTrace(cachedTrace, trace)
         || (shouldRebuildTraceEventsGraph && shouldRebuildTraceEntitiesGraph)) {
       Instant start = Instant.now();
-      StructuredTraceGraph graph = StructuredTraceGraph.createGraph(trace);
+      StructuredTraceGraph graph = new StructuredTraceGraph(trace);
       LOG.debug(
           "Time taken in building StructuredTraceGraph, duration_millis:{} for tenantId:{}",
```

Comment:
not related to this PR, but wrap the below debug statement inside a LOG.isDebugEnabled() to avoid the computation of `Duration.between(start, Instant.now()).toMillis()`

Reply:
Done